### PR TITLE
FloodScan Analysis for Ethiopia: Gu (AMJ) Season

### PR DESCRIPTION
**Pull request by PaulineNimo**

### CODE REVIEW

**Comment by PaulineNimo** [comment]
Just a note, let's explicitly use `dplyr::select` in the code.

File: make_ethiopia_flood_raster.qmd
Comment (zackarno) [inline make_ethiopia_flood_raster.qmd:1]: didn't i make this file in a previous commit that should already be on main? not sure why its showing to be reviewed?

File: flood/eth/make_ethiopia_flood_raster.qmd
Comment (zackarno) [inline flood/eth/make_ethiopia_flood_raster.qmd:1]: this should be from previous commit?

File: flood_target_store/.gitignore
Comment (zackarno) [inline flood_target_store/.gitignore:1]: did you tel it not to ignore meta folder or was this something i did earlier - i think it should be ignored

File: flood_target_store/meta/meta
Comment (zackarno) [inline flood_target_store/meta/meta:1]: think should be ignored

File: floodscan_susceptibliity_target_pipeline.R
Comment (zackarno) [inline floodscan_susceptibliity_target_pipeline.R]: this table looks problematic duplicate rows with now differentiating id/label (see related [comment](https://github.com/OCHA-DAP/pa-rosea-support/pull/19#discussion_r1569493524)) , Maybe https://github.com/OCHA-DAP/pa-rosea-support/pull/19#discussion_r1569495848 would help?

File: floodscan_susceptibliity_target_pipeline.R
Comment (zackarno) [inline floodscan_susceptibliity_target_pipeline.R]: why split - its just ethiopia

File: floodscan_susceptibliity_target_pipeline.R
Comment (zackarno) [inline floodscan_susceptibliity_target_pipeline.R]: table is missing row identifier. Some of the row numbers represent AMJ numbers while others do not, but its not clearly marked in the gt table. Maybe [this](https://github.com/OCHA-DAP/pa-rosea-support/pull/19#discussion_r1569495848) would help?

File: floodscan_susceptibliity_target_pipeline.R
Comment (zackarno) [inline floodscan_susceptibliity_target_pipeline.R]: you may want to set `.default=NA_character_` and then filter it out before making the `{gt}` tables? currently it's left in, but not clearly marked in the tables as to what it is

File: floodscan_susceptibliity_target_pipeline.R
Comment (zackarno) [inline floodscan_susceptibliity_target_pipeline.R]: what does `_ssn` stand for? -- maybe add `@roxygen` to the function

File: floodscan_susceptibliity_target_pipeline.R
Comment (zackarno) [inline floodscan_susceptibliity_target_pipeline.R]: can just do `lr_wp$Ethiopia`

File: R/tar_flood_exposure.R
Comment (zackarno) [inline R/tar_flood_exposure.R]: this is not actually binarizing floodscan like i did in SOM HRP methodology. We should check if thats actually what we want to do here. This masking step is not actually doing anything.

To binarize we would want to just run: 

```
r_expsure = lgl_thresh_mask * r_wp
```

i guess this is wrong in some of the previous `_targets` i had made as well, but I just checked in the SOM repo and it is correct in there: 

https://github.com/OCHA-DAP/ds-som-risk-analysis-support/blob/125e09745ade3a84a8efbbb5108b816ba590e747/R/tar_flood_exposure.R#L116-L117

**Review by zackarno — CHANGES_REQUESTED** [review]
Hi - nice work!

- I've made some comments, but the main comment is the binarizing step which is not working as it does in the SOM HRP repo. See comment [here](https://github.com/OCHA-DAP/pa-rosea-support/pull/19#discussion_r1569524786). When you fix this, perhaps you can adjust in upstream targets as well.
- I'm not sure whats happened in the git history because it is showing alot of the files i wrote code for under the "files changed" (i.e [comment below](https://github.com/OCHA-DAP/pa-rosea-support/pull/19#discussion_r1569267250))... perhaps a rebase is needed? or if it's easier you could direct me to the exact files/code to review
- We might consider refactoring here or in a separate branch to reduce duplication in targets

File: make_ethiopia_flood_raster.qmd
Comment (PaulineNimo) [reply make_ethiopia_flood_raster.qmd:1]: this is just for Ethiopia. I do not need it, it seems.

File: flood/eth/make_ethiopia_flood_raster.qmd
Comment (PaulineNimo) [reply flood/eth/make_ethiopia_flood_raster.qmd:1]: no, no. this is a new file.

File: flood_target_store/.gitignore
Comment (PaulineNimo) [reply flood_target_store/.gitignore:1]: yeah, should be ignored. not sure why it wasn't.

File: floodscan_susceptibliity_target_pipeline.R
Comment (PaulineNimo) [reply floodscan_susceptibliity_target_pipeline.R]: but this would make it easier to add countries in the future

File: floodscan_susceptibliity_target_pipeline.R
Comment (PaulineNimo) [reply floodscan_susceptibliity_target_pipeline.R]: just a short way of writing season

File: floodscan_susceptibliity_target_pipeline.R
Comment (PaulineNimo) [reply floodscan_susceptibliity_target_pipeline.R]: I fixed it in the last push. I had filtered out for the AMJ season.

File: floodscan_susceptibliity_target_pipeline.R
Comment (PaulineNimo) [reply floodscan_susceptibliity_target_pipeline.R]: I will add it when I look at the other seasons

File: floodscan_susceptibliity_target_pipeline.R
Comment (PaulineNimo) [reply floodscan_susceptibliity_target_pipeline.R]: Future proofing

File: R/tar_flood_exposure.R
Comment (PaulineNimo) [reply R/tar_flood_exposure.R]: So, the masked raster is the same as the unmasked one. I can refactor this better when I get back.

**Comment by PaulineNimo** [comment]
Won't merge now and will leave this open.
- process FS data by country
- add function to assign a season name to dates from `zonal_pop_exposure`

**Comment by PaulineNimo** [comment]
I have made additional changes including changing the admins.

**Comment by PaulineNimo** [comment]
I also added some maps for 5 thresholds. Based on the output, I am leaning more towards the 0.5% threshold. Let me know your thoughts @zackarno

File: R/tar_flood_exposure.R
Comment (zackarno) [inline R/tar_flood_exposure.R:25]: i've improved this code here: 
https://github.com/OCHA-DAP/ds-floodscan-ingest/blob/migrate-to-blob/R/fs_load_utils_tidync.R

Sometimes depending on the boundaries the raster would previously get a bit stretched. Working on making some better functions in order to convert FS to COGs and put on blob as well, but not ready yet.

You will see the resolution is not perfect:

```
resolution  : 0.08295626, 0.08291246  (x, y)
```

File: R/tar_flood_exposure.R
Comment (zackarno) [inline R/tar_flood_exposure.R]: nice these are left over from a while back -- can delete at some point

File: R/tar_flood_exposure.R
Comment (zackarno) [inline R/tar_flood_exposure.R:113]: I think it would be easier if we just end the floodscan lookup after mutating `fs_year`

```r
floodscan_lookup <-  function(r_fs){
  # make a lookup table to be used for raster manipulations
  fs_mos<- floor_date(as_date(names(r_fs)),"month")

 tibble(
    fs_name = as_date(names(r_fs))
  ) |>
  mutate(
      fs_mo = floor_date(fs_name,"month"),
      fs_yr = floor_date(fs_name,"year")
    )
}
```

The additional processing for custom seasons can be added on as separate steps/functions...i.e filtering by seasons, summarising by season. That way this function will remain useful across projects. You will see this is what I did in the linked repo I shared above

File: flood/eth/FloodScan_Exposure_Estimates_ETH_AMJ.qmd
Comment (zackarno) [inline flood/eth/FloodScan_Exposure_Estimates_ETH_AMJ.qmd:1]: doc looks nice, I have the targets running now and will go through it once finished

File: floodscan_susceptibliity_target_pipeline.R
Comment (zackarno) [inline floodscan_susceptibliity_target_pipeline.R]: you realize that binary is set to `FALSE`, but you are calling the output target `*_stats_binary`.

I think you should clear this up in the function... adding more details on the function below
https://github.com/OCHA-DAP/pa-rosea-support/pull/19#discussion_r1602332452

File: R/tar_flood_exposure.R
Comment (zackarno) [inline R/tar_flood_exposure.R:70]: you should include this purrr::map iteration in the binarize_floodscan section as well if you really want the binary thresholds to be implemented as suggested in the target name `*_stats_binary`.

In the section where it's not making a binary raster:

```
if(!binarize_floodscan) {
... 
} 
```
all it's doing is doing is setting values < theshold to 0 and retaining original raw values above that. Therefore, you would want to add this iteration step to the section that is doing the binary reclassification

```
if(binarize_floodscan) {
 ...
}
```

I think what could have happened is that I started to write this `map()` loop at some point during the Somalia risk work, but we never ended up using it because we didn't test various thresholds, rather just went with 20 % flood fraction.

Comment (zackarno) [reply R/tar_flood_exposure.R:70]: so the pseudo code would be like:

```
if(binarize_floodscan){
    if (length(flood_frac_thresh)>1){
        flood_frac_thresh |>
            map( \(thresh_tmp){
                r_binary_tmp  <- ifel(r_fs_resampled>=thresh_tmp,1,0) 
                r_exposure <- r_binary_tmp * r_wp

               # now run zonal stats on R exposure

                })
}
               
# then for not binarizing (if you even want this)

if(!binarize_floodscan){
    if (length(flood_frac_thresh)>1){              
       flood_frac_thresh |>
            map( \(thresh_tmp){
                r_thresholded_tmp  <- ifel(r_fs_resampled>=thresh_tmp,r_fs_resampled,0) 
                r_exposure <- r_thresholded_tmp * r_wp 
    # etc. etc.


  lgl_thresh_mask <- ifel(r_fs_resampled>=flood_frac_thresh,1,0) 
   r_fs_resampled_masked <- mask(r_fs_resampled,lgl_thresh_mask)
   # r_exposure <- r_fs_resampled_masked * r_wp
   r_exposure <- lgl_thresh_mask * r_wp
   ... 

  zonal_stats ... 
}
```

**Comment by zackarno** [comment]
Hey @PaulineNimo - nice work -  got tons of analysis in here!

I think you should have a look specifically at the comments below and see if they make sense to you to address in the code. If what I'm saying here is accurate than the results might be slightly different. Therefore,  if you can address these first I will re-run the targets and continue the review so that the adjustments will be reflected in the notebooks

https://github.com/OCHA-DAP/pa-rosea-support/pull/19#discussion_r1602273467
https://github.com/OCHA-DAP/pa-rosea-support/pull/19#discussion_r1602332452
https://github.com/OCHA-DAP/pa-rosea-support/pull/19#discussion_r1602338784

^ it's weird how when I copy an issue comment link they all end up labelled with the same text, but they are different if you click on them

**Pull request merged**
